core/cap-0046-06.md

```diff
@@ -354,6 +393,56 @@ differently even though they have the same format.
+enabled. This matches the clawback behavior on Stellar classic.
+
+### Balances are u128s
+
+The options considered were u63 (contained in an ScVal), u64, u128, u256, and an
+arbitrary precision big integer. Big integer is unnecessary since u256 is more
+than enough to represent any reasonable balance. This is made more obvious when
+looking at other chains and what they use (Solana uses u64, Near uses u128, and
+ERC-20 uses u256). Note that a u63 is more efficient than u64 in Soroban because
+the u63 can be contained in a 64 bit `ScVal`, while u64 will need be to an
+`ScObject`, which only exists in the host and needs to be referenced by the guest.
+
+Now the question is what should it be instead? u63 is the most efficient with
+u64 close behind, but not the most flexible if you use more than a couple
+decimal places. For example, a u64 balance with 9 decimal places would have a
+max value of 18,446,744,073.709551615. Stellar classic uses a signed 64 bit
+integer (which would be a u63 in Soroban) with seven decimal places, leading to
+a max value of 922,337,203,685.4775807. These values might be sufficient in most
+cases, but there are some edge cases where these limits are too small. For
+example, the US Treasury has a cash balance of $636 billion. This barely works
+in the Stellar case, but isn’t far off from reaching the max. It works in the
+variable decimals case if you give up some decimals. There are other currency
+and asset examples that exceed these limits as well. Instead of forcing issuers
+to compromise on decimals, the best option is to use u128 balances.
+
+u128 gives users more flexibilty than Stellar classic without compromising too
+much in terms of performance. The performance implications of using u128 instead
+of u63 or u64 should be measured.
+
+### No approve function
+This proposal does not include an `approve` function like the one specified in EIP-20. This was done to mitigate the allowance vulnerability specified in this [google doc](https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit). TLDR: If user A specified an allowance of 10 for B, then tries to reduce it to 5, B could spend the allowance before the reduction is accepted on the ledger, and then have an additional 5 to spend. Instead of reducing the allowance to 5, the user actually increased it to 15!
+
+Instead of `approve`, this proposal specifies two functions to deal with allowances - `increase_allowance` and `decrease_allowance`.
+
+### No explicit functions to burn
+This proposal does that include a `burn` or `burnFrom` function that allows a user to burn a balance that it holds. This functionality can be imitated by just sending the balance to the zero ed25519 identifier. The alternative would be to implement `burn` and `burnFrom` similar to what the [OpenZeppelin ERC-20](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20Burnable) implementation does.  
```

Comment:
This is a bit confusing. Wasn't `burn` we had just renamed to `clawback`? We never had a burn function that simply burned the balance from your own account...

Reply:
I think the typo confused you (that -> not). We don't have a burn function to burn your own balance, but it's something that's available in the that specified implementation of ERC-20 and EIP-777.

Reply:
I see, so it seems like we had `burn` with clawback semantics that was different from EIP-777 semantics. Then renaming makes more sense.